api/dao/pinDAO.js

```diff
@@ -0,0 +1,218 @@
+const _          = require('lodash');
+const defaultLog = require('winston').loggers.get('default');
+const mongoose   = require('mongoose');
+const Utils      = require('../helpers/utils');
+const projectDAO = require('../dao/projectDAO');
+
+exports.getProjectPins = async function(user, roles, project, pageNumber, pageSize, sortBy)
+{
+    let skip = null, 
+        limit = null, 
+        sort = null;
+
+    let query = {};
+
+    _.assignIn(query, { "_schemaName": "Project" });
+
+    let fields = ['_id', 'pins', 'name', 'website', 'province', 'pinsRead'];
+
+    // Getting a single project
+    _.assignIn(query, { _id: mongoose.Types.ObjectId(project._id) });
+
+    var data = await Utils.runDataQuery('Project',
+                                        roles,
+                                        query,
+                                        fields, // Fields
+                                        null, // sort warmup
+                                        null, // sort
+                                        null, // skip
+                                        null, // limit
+                                        false, // count
+                                        null,
+                                        true,
+                                        null);
+
+    _.assignIn(query, { "_schemaName": "Organization" });
+
+    let thePins = [];
+    
+    if (data.length === 0) 
+    {
+        // no pins, return empty result;
+        return [{ total_items: 0 }];
+    } 
+    else 
+    {
+        if (data[0].pinsRead && !data[0].pinsRead.includes("public") && user === "public") 
+        {
+          // This is the case that the public api has asked for these pins but they are not published yet
+          return [{ total_items: 0 }];
+        }
+
+        data[0].pins.map(pin => 
```

Comment:
`map` returns an array. 

`thePins = data[0].pins.map(pin => mongoose.Types.ObjectId(pin)) `

Reply:
Copy paste from the original pin code that I hadn't changed. I'll update this and review some of the other original source that was copied that could use some fixes